packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -86,15 +90,93 @@ const VeniaAdapter = props => {
         return null;
     }
 
+    let storeCodeRouteHandler = '';
+    const browserRouterProps = {};
+    if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+        const storeCode = storage.getItem('store_view_code') || STORE_VIEW_CODE;
+        browserRouterProps.basename = `/${storeCode}`;
+        const storeCodes = [];
+        const storeCurrencies = {};
+        if (Array.isArray(AVAILABLE_STORE_VIEWS)) {
+            AVAILABLE_STORE_VIEWS.forEach(store => {
+                storeCodes.push(store.code);
+                storeCurrencies[store.code] =
+                    store.default_display_currency_code;
+            });
+        }
+
+        /**
+         * The current store code won't be matched as it's included as the basename, if we match another store
+         * we need to change the current store.
+         */
+        storeCodeRouteHandler = (
+            <Switch>
+                <Route path={`/:storeCode(${storeCodes.join('|')})?`}>
+                    {({ match }) => {
+                        if (
+                            match.params.storeCode &&
+                            storeCodes.includes(match.params.storeCode)
+                        ) {
+                            /**
+                             * Only execute if one store code is present in the URL, multiple store codes will cause
+                             * the store state to break and cause weird side effects for the user
+                             */
+                            const regex = new RegExp(
+                                `(${storeCodes.join('|')})\/`,
+                                'g'
+                            );
+                            const storeCodesInUrl = window.location.pathname.match(
+                                regex
+                            ).length;
```

Comment:
This throws an error if I navigate to `<url>/fr` with a cold cache. For what it's worth, if you first navigate to `<url>/fr/` and then remove the slash it does not throw an error. Seems to conflict with your second set of verification steps.

Reply:
Updated the regex pattern to search for a slash in front of the code, as that's guaranteed.